crates/ln-dlc-node/src/fee_rate_estimates.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::Result;
+use autometrics::autometrics;
+use bdk::FeeRate;
+use lightning::chain::chaininterface::ConfirmationTarget;
+use lightning::chain::chaininterface::FeeEstimator;
+use lightning::chain::chaininterface::FEERATE_FLOOR_SATS_PER_KW;
+use std::collections::HashMap;
+use tokio::sync::RwLock;
+
+pub struct FeeRateEstimates {
+    client: esplora_client::AsyncClient,
+    fee_rate_cache: RwLock<HashMap<ConfirmationTarget, FeeRate>>,
+    fallbacks: RwLock<FeeRateFallbacks>,
+    runtime_handle: tokio::runtime::Handle,
+}
+
+#[derive(Debug)]
+pub struct FeeRateFallbacks {
+    pub normal_priority: u32,
+    pub high_priority: u32,
+}
+
+impl Default for FeeRateFallbacks {
+    fn default() -> Self {
+        Self {
+            normal_priority: 2000,
+            high_priority: 5000,
+        }
+    }
+}
+
+impl FeeRateEstimates {
+    pub fn new(esplora_url: String, runtime_handle: tokio::runtime::Handle) -> Self {
+        let client = esplora_client::AsyncClient::from_client(esplora_url, reqwest::Client::new());
+
+        let fee_rate_cache = RwLock::new(HashMap::default());
+        let fallbacks = RwLock::new(FeeRateFallbacks::default());
+
+        Self {
+            client,
+            fee_rate_cache,
+            fallbacks,
+            runtime_handle,
+        }
+    }
+
+    pub(crate) fn get(&self, target: ConfirmationTarget) -> FeeRate {
+        let (cache, fallbacks) = self.runtime_handle.block_on(async {
+            (
+                self.fee_rate_cache.read().await,
+                self.fallbacks.read().await,
+            )
+        });
+
+        cache.get(&target).copied().unwrap_or_else(|| {
+            let fee_rate = match target {
+                ConfirmationTarget::Background => FEERATE_FLOOR_SATS_PER_KW,
+                ConfirmationTarget::Normal => fallbacks.normal_priority,
+                ConfirmationTarget::HighPriority => fallbacks.high_priority,
+            };
+            FeeRate::from_sat_per_kwu(fee_rate as f32)
+        })
+    }
+
+    #[autometrics]
+    pub(crate) async fn update(&self) -> Result<()> {
+        let mut locked_fee_rate_cache = self.fee_rate_cache.write().await;
+
+        let estimates = self.client.get_fee_estimates().await?;
```

Comment:
this one should IMHO be in spawn blocking, as it will take some time to finish.

source: https://kerkour.com/rust-async-await-what-is-a-runtime 
```
[...]
What does it mean? Not calling functions that may run for more than [10 to 100 microseconds](https://ryhl.io/blog/async-what-is-blocking/) directly. Instead, spawn_blocking them.
```

Reply:
the point is that to do this judiciously, if it can block for longer than the time above.
 there's no harm to spawning it on a worker thread instead of the main runtime.

Reply:
> this one should IMHO be in spawn blocking, as it will take some time to finish.

This call doesn't do anything other than build and send the HTTP request and it `await`s for all the expensive bits: building, sending and deserialising.

`tokio::spawn_blocking` targets expensive non-`async` code: we won't even be able to call `get_fee_estimates` within the closure!